rails/planit/app/controllers/home_controller.rb

```diff
@@ -1,5 +1,11 @@
 class HomeController < ApplicationController
 
   def index
+    @providers = []
+    @categories = Category.all
+    if params[:category]
+      @providers = (Provider.all).select { |prov| prov.category_id == params[:category].to_i }
```

Comment:
with `Provider.all` you are grabbing all providers from the database and loading them in memory. With the `select` you are filtering the elements in memory. Imagine if you had 1000+ providers, that's really innefficient.
What you want is to query in DB the Providers with the specific category.
```
Provider.where(category_id: params[:category].to_i)
```